Add onReset option to contribution filter

andrewbiang888 opened the pull request:
🥳

andrewbiang888 committed:
add new filter component

andrewbiang888 committed:
Merge branch 'develop' of https://github.com/hackoregon/openelections into feature/contribution-search
Pre merge commit

andrewbiang888 committed:
hook up api redux duck and do basic style tweek

andrewbiang888 committed:
hook up api redux duck and do basic style tweek

billoosijok committed:
Created DateRange field and added to the contributions filter

billoosijok committed:
Created DateRange field and added to the contributions filter

billoosijok committed:
WIP: DateRangeField

billoosijok committed:
WIP: DateRangeField

andrewbiang888 committed:
hack stylying for select data picker

andrewbiang888 committed:
hack stylying for select data picker

billoosijok committed:
Getting closer to making the date range field work

billoosijok committed:
Getting closer to making the date range field work

billoosijok committed:
Implemented DateTimeRange field

billoosijok committed:
Implemented DateTimeRange field

billoosijok committed:
Implemented DateTimeRange field

billoosijok committed:
Implemented DateTimeRange field

billoosijok committed:
GH-492 Implemented contributions filter

billoosijok committed:
GH-492 Implemented contributions filter

billoosijok committed:
Merge branch 'develop' of https://github.com/hackoregon/openelections into feature/contribution-search

billoosijok committed:
Bug fixes

billoosijok committed:
Bug fixes

kdv24 committed:
add initial phone masking

kdv24 committed:
WIP- add validation masking for currency

- currently not sure how to implement with formik and the handleChange method
- do we want two decimal points or just commas or ...?
- what should the initial value be?

kdv24 committed:
add initial email validation masking

- to test, add EmailField to Contributions Section that currently uses TextField
- logic needs to be added to validate for Email or Phone depending on ContactType selected
- same as currency validation, I'm not sure handleChange and values are okay with Formik

kdv24 committed:
add initial phone field validation masking- same issues as currency & email

kdv24 committed:
temporary change to ContributionSections for testing validation masking:

- currency
- email
- phone

- needs logic for email vs. phone validation scenarios

kdv24 committed:
update contactType & contactInfo info fields to phone and email

kdv24 committed:
update formFields from contactType/contactInfo to phone and email

kdv24 committed:
remove contactType and contactInformation from api.js

- I believe all the enums and logic I took out is no longer necessary, but would
appreciate another set of eyes on this.

kdv24 committed:
update email and phone fields to use validation masking

kdv24 committed:
update amountOfContribution to undefined

punkael committed:
Expose formik, test if amount is over 500 and change submitForMatch to Yes

kdv24 committed:
change amountOfContribution & checkNumber from undefined to  empty strings

punkael committed:
Fixes console error, sets default to No

punkael committed:
Adds set submitForMatch based on type, subtype, and contributor type

punkael committed:
Merge branch 'develop' into feature/issue-GH-645

kdv24 committed:
add in phoneType select field

billoosijok committed:
Add reset feature

billoosijok committed:
Add reset feature

andrewbiang888 committed:
tweak the reset button

andrewbiang888 committed:
tweak the reset button

andrewbiang888 committed:
merge in develop

andrewbiang888 committed:
fix package.json merge conflict

andrewbiang888 committed:
fix package.json merge conflict

danmelton commented on app/src/components/Fields/TimeField.js:
```diff
@@ -3,10 +3,16 @@ import PropTypes from "prop-types";
   return (
   <TextFieldMaterial
     required={isRequired}
+>>>>>>> 7a9edf7cd01fab28342d7d547d47ffdbd7325a7a
```
ekk, looks like merge hell happened. Recommend doing a new branch off of develop, and cherry-pick your changes into it.

kdv24 committed:
remove labels and add label from forms (#658)

- removed from displaying on table header and form sections
- removed from displaying on both AddExpense and Contributions

punkael committed:
Merge branch 'develop' into feature/issue-GH-645

punkael committed:
Fixes logic, uses options text to compare

kdv24 committed:
WIP- current attempt a at getting formik to work with masking library

kdv24 committed:
update values in api and contributionFields

kdv24 committed:
Merge pull request #654 from hackoregon/feature/issue-GH-645

Feature/issue Fixes #645

punkael committed:
Pass in field name to onchange

kdv24 committed:
pass id directly to formik onChange method for state change

kdv24 committed:
prevent adding more than two decimal places

kdv24 committed:
Update contact types - closes 587 (#653)

* update contactType & contactInfo info fields to phone and email

* update formFields from contactType/contactInfo to phone and email

* remove contactType and contactInformation from api.js

- I believe all the enums and logic I took out is no longer necessary, but would
appreciate another set of eyes on this.

* fix date formatting

* add in phoneType select field

* update values in api and contributionFields

* mapped phoneType to post fields

kdv24 committed:
fix conflicts when rebased

kdv24 committed:
Merge pull request #650 from hackoregon/feature/add-validation-mask-GH-589

Feature/add validation mask gh 589

danmelton committed:
working through geocoding

danmelton committed:
working through geocoding

danmelton committed:
adds matchId for govadmins to get contributions closes #618

danmelton committed:
adds geocoding service to contribution closes #290

danmelton committed:
adds geocoding service to contribution closes #290

ZhangWS committed:
Resolved #326, ,#381, #382 (#536)

* Added geocode-checking

* Add PDX city boundaries data

* #326 Fixing broken tests and lint issues

* #326 Added some example text

* #326 Removed Portland zip code list

* cleaned up geolocation code

* #326 fixed lint issue

* Update database.py

danmelton committed:
reversing order of lat lon

danmelton committed:
integrating geolocation lookup for data science

danmelton committed:
mrege

danmelton committed:
adding boolean to eligible address

danmelton committed:
adds id to params type

danmelton committed:
adds id to params helper

danmelton committed:
fixing typing

danmelton committed:
fix typing

danmelton committed:
wraps up tests

danmelton committed:
adds oaeType to api contributions closes #659 (#664)

* adds oaeType to api contributions closes #659

* adds oaeType to api contributions closes #659

kdv24 committed:
Add oae field - closes GH #660 (#665)

* add oae contribution type where missing

* add oae contribution type to api.js

* update incorrect naming

* fix enum names & cases

* fix UI values for OaeType

danmelton committed:
fixes all the tests

danmelton committed:
merge

danmelton committed:
Merge branch 'develop' of github.com:hackoregon/openelections into develop

danmelton committed:
fixing test

punkael committed:
Fixes gh 640 (#663)

* Added field called entityName that passes it's value to lastName
Dynamically require first and last name vs entityName
Removed remarks that made post fields static
Moved type of contributor under contributor

* Post entityName to name

danmelton committed:
addingjobs scripts

danmelton committed:
Merge branch 'develop' of github.com:hackoregon/openelections into develop

andrewbiang888 committed:
rebase develop

andrewbiang888 committed:
rebasing and etc
